src/open_inwoner/accounts/views/contactmoments.py

```diff
@@ -70,84 +59,53 @@ def handle_no_permission(self):
+    ) -> FetchParameters | None:  # noqa: E704
+        ...
 
 
 class KlantContactMomentBaseView(
     CommonPageMixin, BaseBreadcrumbMixin, KlantContactMomentAccessMixin, TemplateView
 ):
-    def get_kcm_data(
-        self,
-        kcm: KlantContactMoment,
-        local_kcm_mapping: dict[str, KlantContactMomentAnswer] | None = None,
-    ) -> KCMDict:
-        data = {
-            "registered_date": kcm.contactmoment.registratiedatum,
-            "channel": kcm.contactmoment.kanaal.title(),
-            "text": kcm.contactmoment.tekst,
-            "url": reverse("cases:contactmoment_detail", kwargs={"kcm_uuid": kcm.uuid}),
-            # eSuite extra
-            "identificatie": kcm.contactmoment.identificatie,
-            "type": kcm.contactmoment.type,
-            "status": Status.safe_label(kcm.contactmoment.status, _("Onbekend")),
-            "antwoord": kcm.contactmoment.antwoord,
-            "new_answer_available": contactmoment_has_new_answer(
-                kcm.contactmoment, local_kcm_mapping=local_kcm_mapping
-            ),
-        }
-
-        data["onderwerp"] = self.get_kcm_subject(kcm)
-
-        return data
+    def get_service(self) -> VragenService:
+        # TODO: Refactor to support both OpenKlant2 and eSuite services at once
+        return eSuiteVragenService()
 
     def get_context_data(self, **kwargs):
         ctx = super().get_context_data(**kwargs)
         ctx["anchors"] = self.get_anchors()
         return ctx
 
-    def get_kcm_subject(
-        self,
-        kcm: KlantContactMoment,
-    ) -> str | None:
-        """
-        Determine the subject (`onderwerp`) of a `KlantContactMoment.contactmoment`:
-            1. replace e-suite subject code with corresponding OIP configured subject or
-            2. return the first OIP subject if multiple subjects are mapped to the same
-               e-suite code or
-            3. return the the e-suite subject code if no mapping exists
-        """
-        e_suite_subject_code = getattr(kcm.contactmoment, "onderwerp", "")
-
-        try:
-            subject = ContactFormSubject.objects.get(subject_code=e_suite_subject_code)
-        except ContactFormSubject.MultipleObjectsReturned as exc:
-            logger.warning(
-                "Multiple OIP subjects mapped to the same e-suite subject code for ",
-                "contactmoment %s; using the first one",
-                kcm.contactmoment.url,
-                exc_info=exc,
-            )
-            return ContactFormSubject.objects.first().subject
-        except ContactFormSubject.DoesNotExist as exc:
-            logger.warning(
-                "Could not determine OIP subject for contactmoment %s; "
-                "falling back on e-suite subject code ('onderwerp')",
-                kcm.contactmoment.url,
-                exc_info=exc,
+    @property
+    def fetch_params(self):
+        if not (
+            fetch_params := self.get_service().get_fetch_parameters(
```

Comment:
We end up building the service multiple times: once explicitly in `get_context_data` of the list view, then again implicitly by accessing `self.fetch_params` (lines 164, 167). Perhaps we can do the `get_service` in the `__init__` of the view to avoid this?